app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -72,7 +72,9 @@ def check_enrollments(enrollments)
 
     enrollments.each do |enrollment|
       # Add a unique ID for enrollments that don't have one
-      enrollment.update(unique_id: enrollment.usps_unique_id) if enrollment.unique_id.blank?
+      if enrollment.unique_id.blank?
+        enrollment.update(unique_id: InPersonEnrollment.generate_unique_id)
+      end
```

Comment:
Note that the [original PR](https://github.com/18F/identity-idp/pull/6623/files#diff-3999219037053b6ad66255d8f7e863be02a97e1543c499ad75f3bd98b28a1ed7R30) set the unique ID to the deprecated `enrollment.usps_unique_id` but my understanding is that we want to move away from using that method and instead use the new `InPersonEnrollment.generate_unique_id` method. Is there any reason to use the deprecated method here?

Reply:
what if we used some hooks on the model, like we do for the [NonNullUuid](https://github.com/18F/identity-idp/blob/main/app/models/concerns/non_null_uuid.rb) concern? something like a `before_save` that checks the UUID in all places so that we don't have to add plugs throughout the codebase?

or run a migration manually and make the column not-null in the DB so we can close this loop?

Reply:
Ah, but the issue with this is that these enrollments would have already been associated with an enrollment in the USPS API, and those enrollments would have used the deprecated unique ID scheme. So we should indeed use the old scheme

Reply:
@zachmargolis Yep that makes sense. Added a hook in 072091dcc562604b4881eb6e74241319f3a6b9fb (removed the unnecessary if condition in a later commit)